model/flow/protocol_state.go

```diff
@@ -90,7 +105,22 @@ func (c *EpochStateContainer) Copy() *EpochStateContainer {
 		SetupID:          c.SetupID,
 		CommitID:         c.CommitID,
 		ActiveIdentities: c.ActiveIdentities.Copy(),
+		EpochExtensions:  clone.Clone(c.EpochExtensions),
+	}
+}
+
+// ExtendEpoch appends an epoch extension to the EpochStateContainer.
+// It is expected that the epoch extension is contiguous with the last extension, otherwise an exception is returned.
+// No errors are expected during normal operation.
+func (c *EpochStateContainer) ExtendEpoch(epochExtension EpochExtension) error {
+	if len(c.EpochExtensions) > 0 {
+		lastExtension := c.EpochExtensions[len(c.EpochExtensions)-1]
+		if lastExtension.FinalView+1 != epochExtension.FirstView {
+			return fmt.Errorf("epoch extension is not contiguous with the last extension")
+		}
 	}
+	c.EpochExtensions = append(c.EpochExtensions, epochExtension)
+	return nil
 }
```

Comment:
I have mixed feelings about this implementation, because there is _no_ consistency check for the first extension. Some thoughts:
* Conceptually, we can only ever extend the _current_ epoch. 
   * Hence, conceptually the method `ExtendEpoch` could be attached to the `EpochStateContainer` or `RichProtocolStateEntry` and we could rename it to `ExtendCurrentEpoch`
   * While `EpochStateContainer` cannot check the consistency of the views for the very first extension, `RichProtocolStateEntry` can! As far as I can tell, this would not break anything, because all the higher-level logic anyway works on the `RichProtocolStateEntry`.
   * Furthermore, we were thinking about: In case of entering EFT when a future epoch is in its setup phase, we want to drop the `next` epoch's  setup data  from the state, because this data is anyway outdated. 

     If we dropped the next epoch's setup data, we would need to modify not only the  `EpochStateContainer` but also the `RichProtocolStateEntry`. Hence, attaching the method `ExtendCurrentEpoch` to the `RichProtocolStateEntry` makes sense, because the `RichProtocolStateEntry` could clean up its own data as well as the data from the underlying `ProtocolStateEntry`